packages/frontend/src/components/back_link.tsx

```diff
@@ -0,0 +1,30 @@
+import { faChevronLeft } from "@fortawesome/free-solid-svg-icons";
+import type { FontAwesomeIconProps } from "@fortawesome/react-fontawesome";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useLocation } from "react-router-dom";
+import type { LocationState } from "../state/nav";
+import { Link } from "./link";
+
+export function BackLink({
+  className,
+  defaultLink,
+  ...props
+}: Omit<Omit<FontAwesomeIconProps, "icon">, "onclick"> & {
+  defaultLink: string;
+}) {
+  const { state } = useLocation() as { state: LocationState };
```

Comment:
I'm not going to request-changes this, but I think `useHistory()` would have been more ideal.